src/main/java/com/google/jenkins/plugins/computeengine/client/ComputeClient.java

```diff
@@ -418,6 +447,94 @@ public void deleteTemplate(String projectId, String templateName) throws IOExcep
         return instanceTemplates;
     }
 
+    /**
+     * Creates persistent disk snapshot for Compute Engine instance.
+     * This method blocks until the operation completes.
+     *
+     * @param projectId Google cloud project id (e.g. my-project).
+     * @param zone Instance's zone.
+     * @param instanceId Name of the instance whose disks to take a snapshot of.
+     *
+     * @throws IOException If an error occured in snapshot creation.
+     * @throws InterruptedException If snapshot creation is interrupted.
+     */
+    public void createSnapshot(String projectId, String zone, String instanceId) throws IOException, InterruptedException {
+        try {
+            zone = nameFromSelfLink(zone);
+            Instance instance = compute.instances().get(projectId, zone, instanceId).execute();
+
+            //TODO: JENKINS-56113 parallelize snapshot creation
+            for (AttachedDisk disk : instance.getDisks()) {
+                    String diskId = nameFromSelfLink(disk.getSource());
+                    createSnapshotForDisk(projectId, zone, diskId);
+            }
+        } catch (InterruptedException ie) {
+            // catching InterruptedException here because calling function also can throw InterruptedException from trying to terminate node
+            LOGGER.log(Level.WARNING,"Error in creating snapshot.", ie);
+            throw ie;
+        } catch (IOException ioe) {
+            LOGGER.log(Level.WARNING,"Interruption in creating snapshot.", ioe);
+            throw ioe;
+        }
+    }
+
+    /**
+     * Given a disk's name, create a snapshot for said disk.
+     *
+     * @param projectId Google cloud project id (e.g. my-project).
+     * @param zone Zone of disk.
+     * @param diskId Name of disk to create a snapshot for.
+     *
+     * @throws IOException If an error occured in snapshot creation.
+     * @throws InterruptedException If snapshot creation is interrupted.
+     */
+    public void createSnapshotForDisk(String projectId, String zone, String diskId) throws IOException, InterruptedException {
+        Snapshot snapshot = new Snapshot();
+        snapshot.setName(diskId);
+
+        Operation op = compute.disks().createSnapshot(projectId, zone, diskId, snapshot).execute();
+        // poll for result
+        waitForOperationCompletion(projectId, op.getName(), op.getZone(), SNAPSHOT_TIMEOUT_MILLISECONDS);
+    }
+
+    /**
+     * Deletes persistent disk snapshot. Does not block.
+     *
+     * @param projectId Google cloud project id (e.g. my-project).
```

Comment:
Repeating "(e.g. my-project)" in the \@param annotation throughout seems unnecessary. ProjectId is a pretty established concept.

Reply:
done